File: BusinessLogic/Questionnaires/Mapping/Model/Cat_Tipo_Preguntas.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Cat_Tipo_Preguntas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_tipo_pregunta { get; set; }
       public string? Tipo_pregunta { get; set; }
       public string? Descripcion { get; set; }
       public string? Estado { get; set; }
       public DateTime? Fecha_crea { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public bool? Editable { get; set; }
       public string? Descripcion_general { get; set; }
       [OneToMany(TableName = "Cat_Valor_Preguntas", KeyColumn = "Id_tipo_pregunta", ForeignKeyColumn = "Id_tipo_pregunta")]
       public List<Cat_Valor_Preguntas>? Cat_Valor_Preguntas { get; set; }
       [OneToMany(TableName = "Pregunta_Tests", KeyColumn = "Id_tipo_pregunta", ForeignKeyColumn = "Id_tipo_pregunta")]
       public List<Pregunta_Tests>? Pregunta_Tests { get; set; }
       
   }
}

File: UI/wwwroot/Questionnaires/FrontModel/Cat_Tipo_Preguntas.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
//@ts-ignore
import { Cat_Valor_Preguntas } from './Cat_Valor_Preguntas.js';
import { Pregunta_Tests } from './Pregunta_Tests.js';
class Cat_Tipo_Preguntas extends EntityClass {
    /** @param {Partial<Cat_Tipo_Preguntas>} [props] */
    constructor(props) {
        super(props, 'EntityQuestionnaires');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }
   /**@type {Number}*/ Id_tipo_pregunta;
   /**@type {String}*/ Tipo_pregunta;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Date}*/ Fecha_crea;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Boolean}*/ Editable;
   /**@type {String}*/ Descripcion_general;
   /**@type {Array<Cat_Valor_Preguntas>} OneToMany*/ Cat_Valor_Preguntas;
   /**@type {Array<Pregunta_Tests>} OneToMany*/ Pregunta_Tests;
}
export { Cat_Tipo_Preguntas };
